plugins/pulp_rpm/plugins/migrations/0042_add_suse_fields_to_errata.py

```diff
@@ -0,0 +1,29 @@
+import logging
+
+from pulp.server.db.connection import get_collection
+from pulp.server.db.migrations.lib import utils
+
+
+_logger = logging.getLogger('pulp_rpm.plugins.migrations.0042')
+
+
+def migrate(*args, **kwargs):
+
+    erratum_collection = get_collection('units_erratum')
+    total_erratum_units = erratum_collection.count(
+        {'relogin_suggested': {'$exists': False}, 'restart_suggested': {'$exists': False}})
+
+    with utils.MigrationProgressLog('Erratum', total_erratum_units) as migration_log:
```

Comment:
I'm not sure why progress shows N out of N units, when `migration_log.progress()
` is called 2N times. I guess it's a bug in progress tooling.

I get that we need two loops to handle the case when migration was interrupted but I guess then we need 2 separate progress logs as well... and the total number of units can potentially be different in such case.

Maybe something like this:
``` 
fields_to_add = ['relogin_suggested', 'restart_suggested']
for field in fields_to_add:
    total_erratum_units = erratum_collection.count(
        {field: {'$exists': False}})

    with utils.MigrationProgressLog('Erratum (%s field)' % field, total_erratum_units) as migration_log:
        for errata in erratum_collection.find(
                {field: {'$exists': False}})...
            ...
```
Thoughts?